Migrate Linux and windows Docker build jobs from CircleCi to GitHub Action.

nitaliya opened the pull request:
Migrate Linux and windows Docker build jobs
Ticket: [OTL-1259](https://signalfuse.atlassian.net/browse/OTL-1259).

Adding GitHub Action jobs of Linux and windows Docker build
Removing CircleCi jobs of Linux and windows Docker build.

jcheng-splunk commented:
@nitaliya Please rebase and delete the circleci config if there are all jobs have been migrated.

nitaliya commented:
> @nitaliya Please rebase and delete the CircleCI config-if all jobs have been migrated.

Hello @jcheng-splunk,

Thanks for the heads up and yes we rebase the PR as a part of the routine steps that we follow on our end. For the multiple PRs that we raise in the same repository when one of those PRs gets approved and merged from your end, all the other open PRs get out of sync and will show behind the PR that got merged.

For this PR, I rebased it this morning. and As all jobs are migrated I have removed the config file of CircleCI. If there are no further changes needed, please approve and merge this PR.

jcheng-splunk commented on .github/workflows/build-and-test.yml:
```diff
@@ -433,3 +433,115 @@ jobs:
+            exit 1
+          fi
+      
+      - name: Save docker image
```
Let's skip saving and archiving the image to save time.

nitaliya replied:
Removed save docker image step for the GitHub Action workflow now.

jcheng-splunk commented on .github/workflows/build-and-test.yml:
```diff
@@ -433,3 +433,115 @@ jobs:
+            exit 1
+          }
+      
+      - name: Save docker image
```
Let's skip saving and archiving the image to save time.

nitaliya replied:
Removed save docker image step for the GitHub Action workflow now.

jcheng-splunk reviewed (approved):
Thanks @nitaliya. We'll need to coordinate to disable the circleci checks and builds before merging.
